operator/pkg/apis/controlplane/v1alpha1/controlplane.go

```diff
@@ -44,7 +45,17 @@ type ControlPlaneList struct {
 type ControlPlaneSpec struct {
 	KubernetesVersion string     `json:"kubernetesVersion,omitempty"`
 	Master            MasterSpec `json:"master,omitempty"`
-	Etcd              *Component `json:"etcd,omitempty"`
+	Etcd              Etcd       `json:"etcd,omitempty"`
+}
+
+type StorageSpec struct {
+	StorageClassName string            `json:"storageClassName,omitempty"`
+	Size             resource.Quantity `json:"size,omitempty"`
+}
+
+type Etcd struct {
+	Component   `json:",inline"`
+	StorageSpec StorageSpec `json:"storageSpec,omitempty"`
```

Comment:
This violates kubernetes patterns to call it `StorageSpec`. Idiomatically, it would be `Storage`

Reply:
Otherwise you end up with a lot of stuttering spec.etcdspec.storagespec.

Reply:
yes you are right. What was I thinking.